api/odigos/v1alpha1/instrumentatioconfig_types.go

```diff
@@ -20,25 +20,30 @@ type InstrumentationConfig struct {
 // with the comments used as the description.
 type InstrumentationConfigSpec struct {
 
-	// free text description of the instrumentation config for humans
-	Name string `json:"name,omitempty"`
+	// Each object describes a single workload configuration
+	Workload Workload `json:"workload"`
+
+	// config for this workload.
+	// the config is a list to allow for multiple config options and values to be applied.
+	// the list is processed in order, and the first matching config is applied.
+	Config []WorkloadInstrumentationConfig `json:"config"`
```

Comment:
Why not use `common.ProgrammingLanguage`  instead of String for `InstrumentationLibrary.Language`